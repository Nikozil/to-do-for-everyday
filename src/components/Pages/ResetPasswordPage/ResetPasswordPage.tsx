import React from 'react';
import { useDispatch } from 'react-redux';
import GreetingsComponent from '../../../common_components/GreetingsComponent/GreetingsComponent';
import { resetPassword } from '../../../Redux/modules/userSlice';
import ResetPasswordForm from '../../Forms/ResetPasswordForm/ResetPasswordForm';

const ResetPasswordPage: React.FC = () => {
  const dispatch = useDispatch();

  const handleResetPassword = async (email: string) => {
    const response = await dispatch(resetPassword(email));
    return response;
  };

  return (
    <GreetingsComponent>
      <ResetPasswordForm handleSubmit={handleResetPassword} />
    </GreetingsComponent>
  );
};

export default ResetPasswordPage;
