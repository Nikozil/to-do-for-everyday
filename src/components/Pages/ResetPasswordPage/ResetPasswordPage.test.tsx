import { render, waitFor } from '../../../utils/tests/test-utils';
import ResetPasswordPage from './ResetPasswordPage';

describe('ResetPasswordPage tests', () => {
  it('render correct', async () => {
    const { asFragment } = render(<ResetPasswordPage />);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
