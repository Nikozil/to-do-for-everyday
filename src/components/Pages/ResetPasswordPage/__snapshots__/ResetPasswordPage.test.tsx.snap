// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordPage tests render correct 1`] = `
<DocumentFragment>
  <div
    class="greeting-page"
  >
    <div
      class="greeting-page__content"
    >
      <a
        href="/"
      >
        <div
          class="greeting-page__logo"
        >
          <img
            alt="Logo"
            src="Logo.png"
          />
        </div>
      </a>
      <form
        action="#"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="email"
          >
            Email
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="guest@gmail.com"
            type="email"
            value=""
          />
          <div
            class="form-text login-form__error"
          />
        </div>
        <div
          class="d-flex flex-column mt-1 "
        >
          <button
            class="btn btn-success center-block "
            type="submit"
          >
            Отправить пароль на почту
          </button>
        </div>
      </form>
    </div>
  </div>
   
</DocumentFragment>
`;
